src/main/java/io/supertokens/webserver/WebserverAPI.java

```diff
@@ -255,6 +255,15 @@ protected AppIdentifierWithStorage getAppIdentifierWithStorage(HttpServletReques
 
         Storage storage = StorageLayer.getStorage(tenantIdentifier, main);
         Storage[] storages = StorageLayer.getStoragesForApp(main, tenantIdentifier.toAppIdentifier());
+
+        // Make the `storage` is the first item in the `storages` array, to prioritize the tenant of interest
+        for (int i = 1; i < storages.length; i++) {
+            if (storages[i] == storage) {
+                storages[i] = storages[0];
+                storages[0] = storage;
+                break;
+            }
+        }
```

Comment:
this should not happen here, but instead in the getStorage function in the appIdentifierWithStorage

Reply:
Done